Extension/src/LanguageServer/configurations.ts

```diff
@@ -2307,15 +2330,17 @@ export class CppProperties {
             return;
         }
         const compileCommandsFile: string | undefined = this.resolvePath(compileCommands);
+        const compileCommandsLastChanged: Date | undefined = this.compileCommandsFileWatcherFallbackTime.get(compileCommandsFile);
         fs.stat(compileCommandsFile, (err, stats) => {
             if (err) {
                 if (err.code === "ENOENT" && this.compileCommandsFile) {
+                    this.compileCommandsFileWatchers.forEach((watcher: fs.FSWatcher) => watcher.close());
                     this.compileCommandsFileWatchers = []; // reset file watchers
                     this.onCompileCommandsChanged(compileCommandsFile);
                     this.compileCommandsFile = null; // File deleted
                 }
-            } else if (stats.mtime > this.compileCommandsFileWatcherFallbackTime) {
-                this.compileCommandsFileWatcherFallbackTime = new Date();
+            } else if (compileCommandsLastChanged !== undefined && stats.mtime > compileCommandsLastChanged) {
```

Comment:
Can you change it to call `this.compileCommandsFileWatcherFallbackTime.get(compileCommandsFile)` in the stat call back? e.g.

```TypeScript
} else {
    const compileCommandsLastChanged: Date | undefined = this.compileCommandsFileWatcherFallbackTime.get(compileCommandsFile);
    if (compileCommandsLastChanged !== undefined && stats.mtime > compileCommandsLastChanged) {
...
```

Reply:
Other than that it seems good to me.

Reply:
@sean-mcmanus fixed